### PR TITLE
refactor anyUnitsAttr - re-examine

### LINKED ISSUE #555: refactor `anyUnitsAttr`
per PR #549 (for #545), consider the change in `nxdlTypes.xsd` of `anyUnitsAttr` from:

	<xs:simpleType name="anyUnitsAttr">
		<xs:restriction base="xs:string">
		</xs:restriction>
	</xs:simpleType>

to:

	<xs:simpleType name="anyUnitsAttr">
		<xs:union memberTypes="xs:string">
		</xs:union>
	</xs:simpleType>

and then converting `<xs:enumeration` to  `<xs:simpleType` for each type of units

**IMPORTANT**  The result of these changes in `nxdlTypes.xsd` should bump the minor version number of the XML Schema.

### PR DESCRIPTION
**Pull request by prjemian**
fixes #555

### CODE REVIEW

**Comment by prjemian** [comment]
need to merge master into this branch

**Comment by prjemian** [comment]
Willing to wait on this until next milestone.  Today's release will then be merged into this PR, making it easier to resolve what actually will be changed.

**Comment by prjemian** [comment]
Can this be merged now?

**Comment by prjemian** [comment]
@PeterC-DLS Could you review this?

File: base_classes/NXtransformations.nxdl.xml
Comment (PeterC-DLS) [inline base_classes/NXtransformations.nxdl.xml:180]: Leave this file unchanged

**Review by PeterC-DLS — CHANGES_REQUESTED** [review]

File: base_classes/NXtransformations.nxdl.xml
Comment (prjemian) [reply base_classes/NXtransformations.nxdl.xml]: done

**Review by PeterC-DLS — APPROVED** [review]

**Comment by prjemian** [comment]
Thanks for the review.  This is not waiting on any vote so I'll merge now so there will not be another merge conflict if any of the existing PRs are merged.  I expect each one of those may need to be rebased to master before they are merged.

**Pull request merged**
